### PR TITLE
M2 settings

### PR DESCRIPTION
**Pull request by arthurkkhi**
Compare to latest M0
8.2% loss in bdrate score
776% increase in speed

### CODE REVIEW

File: Source/Lib/Codec/EbModeDecision.c [hunk 1]
```diff
@@ -2241,7 +2241,7 @@ EbErrorType ProductGenerateMdCandidatesCu(
 #if ENABLE_INTRA_4x4
 #if ENCODER_MODE_CLEANUP
 #if TUNED_SETTINGS_FOR_M1
-        if ((picture_control_set_ptr->enc_mode <= ENC_M1) ||
+        if ( 1 ||
```
Comment (hassount) [inline Source/Lib/Codec/EbModeDecision.c]: can we just remove the 1 ||

Comment (arthurkkhi) [reply Source/Lib/Codec/EbModeDecision.c]: The line is separated by macro, so the full condition statement is actually:

#if TUNED_SETTINGS_FOR_M1
        if ( 1 ||
#else
        if ((picture_control_set_ptr->enc_mode <= ENC_M0) ||
#endif
            (context_ptr->blk_geom->bwidth != 4 && context_ptr->blk_geom->bheight != 4))

If I remove **1 ||** then the check will depend on second statement.
What I can do is to remove the entire **If** statement.

**Comment by EwoutH** [comment]
Against master, the speed of `-enc-mode 2` increased with 15,8% while the bitrate decreased with 3,2%. Unfortunately, I don't have Psnr (or other efficiency metrics) integrated in my test process yet (#97 would help).

|                | df11eb0 | 0f77af8 |
|----------------|--------:|--------:|
| Speed (fps)    |   0,587 |   0,680 |
| Bitrate (kb/s) |  328,63 |  317,98 |

|                | df11eb0 | 0f77af8 |
|----------------|--------:|--------:|
| Speed (fps)    |  100,0% |  115,8% |
| Bitrate (kb/s) |  100,0% |   96,8% |



- **Video:** Morocco (1080p 8-bit 4:2:0)
- **Settings:** `-enc-mode 2 -n 240`
- **System:** i5-4590 (4c/4t, AVX2), 16GB DDR3, Windows 10 64-bit

[Full details](https://docs.google.com/spreadsheets/d/1p3PJQMkyhIrXEL6MBBQiwTnWmGyHY_zuw7_XAD0def4/edit#gid=2045023166).

**Comment by hassount** [comment]
Closing this PR as we found some bugs between the presets. we plan to fix all presets in one upcoming PR

**Pull request closed without merging**